### PR TITLE
subcards may have sub sub cards

### PR DESCRIPTION
**Pull request by khodgkinson-he**
### Types of changes

-   [X] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change
Subcards appearing below parent cards may in turn have sub cards beneath them.  
Iterating through main_card and nesting the sub_card loop therein, will facilitate brachiation across the entire tree (sub_cards are added beneath current main_card to be next iteration).

q.v. https://hedev.visualstudio.com/Inventory/_workitems/edit/49409
       https://hedev.visualstudio.com/Inventory/_workitems/edit/49490

### CODE REVIEW

File: arches/app/search/search_export.py [hunk 1]
```diff
@@ -61,10 +61,9 @@ def __init__(self, search_request=None):
         self.set_precision = GeoUtils().set_precision
 
     def insert_subcard_below_parent_card(self, main_card_list, sub_card_list):
-        for sub_card in sub_card_list:
-            parent_obj = sub_card.nodegroup.parentnodegroup_id
-            for main_card in main_card_list:
-                if main_card.nodegroup_id == parent_obj:
+        for main_card in main_card_list:            
+            for sub_card in sub_card_list:
+                if main_card.nodegroup_id == sub_card.nodegroup.parentnodegroup_id:
```
Comment (aj-he) [inline arches/app/search/search_export.py:66]: Just checking that the `for` iterator on `main_card_list` picks up sub cards inserted into it on line 68?

Comment (khodgkinson-he) [reply arches/app/search/search_export.py:66]: Absolutely, `main_card_list` is added to at index+1, so `sub_card` processed on next subsequent iteration.

(don't like changing current loop list on the fly, but minor refactor for bugfix,, o_0)

Comment (aj-he) [reply arches/app/search/search_export.py:66]: ok

**Review by aj-he — APPROVED** [review]
all good

**Pull request merged**